### PR TITLE
Tcs 21 include the filter that is being used in the return payload

### PR DESCRIPTION
**Pull request by MateaKanjuga**
- Modified subdetails function in segmentedtickerplant.q to include filter used
- Added filtermap.q in code/segmentedtickerplant with function to create a dictionary of tables and filters

### CODE REVIEW

File: code/processes/segmentedtickerplant.q [hunk 1]
```diff
@@ -2,9 +2,7 @@
 // Contains all TP functionality with additional flexibility
 // Configurable logging and subscriptions
 // Default settings create single TP log per table and rolls logs hourly
-// Subscription to a table can be made in two modes - all or filtered
-// All - publish all data for table
-// Filtered - apply filters to published data, filters defined on client side
```
Comment (rian-morgan) [inline code/processes/segmentedtickerplant.q:7]: plz readd the 3 lines above

File: code/processes/segmentedtickerplant.q [hunk 2]
```diff
@@ -26,11 +24,15 @@ if[.stplg.multilog~`custom;
...
-// subscribers who want to replay need this info 
-subdetails:{[tabs;instruments]
- `schemalist`logfilelist`rowcounts`date`logdir!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG)
+
+
+// Subscribers who want to replay need this info
+// Modified to include information about filter used
+subdetails:{[tabs;instruments;segid]
+ `schemalist`logfilelist`rowcounts`date`logdir`filter!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG; .stpps.filtermap\:[tabs;segid])
  }
```
Comment (rian-morgan) [inline code/processes/segmentedtickerplant.q:33]: Will need to wait for Callums PR then move your changes here into his new function segmentedsubdetails

File: code/segmentedtickerplant/filtermap.q [hunk 3]
```diff
@@ -0,0 +1,10 @@
+\d .stpps
+
+// Function to map the where clause from config table extracted by .stpps.segmentfilter function to tablename
+// Allows use of ` as argument for tables
+
+
+filtermap:{[tabs;segid] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;segid]}
+
+\d . 
+
```
Comment (rian-morgan) [inline code/segmentedtickerplant/filtermap.q:10]: can you move this function into segmentedtickerplant/datastripe.q

File: code/processes/segmentedtickerplant.q [hunk 4]
```diff
@@ -26,11 +24,15 @@ if[.stplg.multilog~`custom;
...
 
-// subscribers who want to replay need this info 
-subdetails:{[tabs;instruments]
- `schemalist`logfilelist`rowcounts`date`logdir!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG)
+
+
+// Subscribers who want to replay need this info
+// Modified to include information about filter used
+subdetails:{[tabs;instruments;segid]
+ `schemalist`logfilelist`rowcounts`date`logdir`filter!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG; .stpps.filtermap\:[tabs;segid])
```
Comment (rian-morgan) [inline code/processes/segmentedtickerplant.q:32]: I dont think  ` .stpps.filtermap\:[tabs;segid]` should need to use each-left. The filtermap function should return a dictionary with all tables requested so no need to iterate over tables here.

**Review by rian-morgan — CHANGES_REQUESTED** [review]

File: code/segmentedtickerplant/datastripe.q [hunk 5]
```diff
@@ -17,3 +17,13 @@ initdatastripe:{
...
 if[.ds.datastripe;.proc.addinitlist[(`initdatastripe;`)]];
+
+
+// Function to map the where clause from config table extracted by .stpps.segmentfilter function to tablename
+// Allows use of ` as argument for tables
+
+\d .stpps
+
+filtermap:{[tabs;segid] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;segid]}
+
```
Comment (rian-morgan) [inline code/segmentedtickerplant/datastripe.q:28]: can you move this above the if statement on line 19? This should be defined before the init runs. Im not sure how this even ran.

**Review by rian-morgan — CHANGES_REQUESTED** [review]

**Review by rian-morgan — APPROVED** [review]

**Review by jwrno1 — APPROVED** [review]

**Pull request merged**